### PR TITLE
fix/identity-service/models-fix

### PR DESCRIPTION
**Pull request by Mohammed-Alhams**

### CODE REVIEW

**Review by devsadeq — APPROVED** [review]

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_address/IManageUserAddressUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package org.thechance.service_identity.domain.usecases.user_address
```
Comment (mustafa-sab3alarab) [inline service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_address/IManageUserAddressUseCase.kt:1]: you can put the kt file on usecases directly

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_management/IUserManagementUseCase.kt [hunk 2]
```diff
@@ -1,12 +1,24 @@
 package org.thechance.service_identity.domain.usecases.user_management
```
Comment (mustafa-sab3alarab) [inline service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_management/IUserManagementUseCase.kt:1]: the same here

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_address/IManageUserAddressUseCase.kt (hunk 1 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_address/IManageUserAddressUseCase.kt:1]: Ok, i will refactor it

File: service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt [hunk 3]
```diff
@@ -1,48 +1,44 @@
...
 
     route("/dashboard/user") {
 
-        get("/{id}") {
-            val id = call.parameters["id"] ?: ""
-            userManagement.getUserById(id)
-        }
-
         get {
             val fullName = call.parameters["full_name"] ?: ""
```
Comment (mustafa-sab3alarab) [inline service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:20]: should throw like another routes

Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:20]: I will work on it

Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:20]: It missed it's bus😁

File: service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt [hunk 4]
```diff
@@ -1,48 +1,44 @@
...
             val username = call.parameters["username"] ?: ""
-            val users = userManagement.getUsersList(fullName, username)
+            val users = userManagement.getUsers(fullName, username)
```
Comment (mustafa-sab3alarab) [inline service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:22]: do mapping here and send http code

File: service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt [hunk 5]
```diff
@@ -1,48 +1,44 @@
...
         }
 
         post("/{id}/permission") {
-            val id = call.parameters["id"] ?: ""
-            val permissionId = call.parameters["permission_id"]?.toInt()
+            val id = call.parameters["id"] ?: throw MissingParameterException(INVALID_REQUEST_PARAMETER)
+            val permissionId = call.receiveParameters()["permission_id"]?.toInt()
                 ?: throw MissingParameterException(INVALID_REQUEST_PARAMETER)
             val result = userManagement.addPermissionToUser(id, permissionId)
             call.respond(result)
```
Comment (mustafa-sab3alarab) [inline service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:37]: send code with respond

File: service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt (hunk 4 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:22]: I haven't got it, can you explain please?

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package org.thechance.service_identity.endpoints.validation
+package org.thechance.service_identity.domain.usecases.util
```
Comment (mustafa-sab3alarab) [inline service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt:2]: remove unnecessary codes and make it in order

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt [hunk 7]
```diff
@@ -1,4 +1,4 @@
...
```
Comment (mustafa-sab3alarab) [inline service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt:1]: change file name please

File: service_identity/src/test/kotlin/org/thechance/service_identity/domain/usecases/validation/UserValidationUseCaseTest.kt [hunk 8]
```diff
@@ -7,7 +7,7 @@ import org.junit.jupiter.api.Test
 
 class UserValidationUseCaseTest {
 
-    private val userValidationUseCase = UserValidationUseCase()
+    private val userValidationUseCase = ValidateUserInfoUseCase()
 
     @Test
     fun should_ReturnFalse_When_TheUsernameIsBlank() {
```
Comment (mustafa-sab3alarab) [inline service_identity/src/test/kotlin/org/thechance/service_identity/domain/usecases/validation/UserValidationUseCaseTest.kt:13]: Ensuring consistent naming of test functions, compare with another file

File: service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt (hunk 4 above)
Comment (mustafa-sab3alarab) [reply service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:22]: val users = userManagement.getUsers(fullName, username).toDto()
            call.respond(HttpStatusCode.OK, users)

**Review by mustafa-sab3alarab — APPROVED** [review]
well done

**Review by mustafa-sab3alarab — DISMISSED** [review]
Refine test functions

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt (hunk 7 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt]: What would be the appropriate alteration?, How can the name be modified without changing its meaning?

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt (hunk 6 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/util/error_codes.kt]: Done

File: service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt (hunk 4 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt:23]: got it

File: service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt (hunk 5 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/endpoints/userManagementRoutes.kt]: Done

File: service_identity/src/test/kotlin/org/thechance/service_identity/domain/usecases/validation/UserValidationUseCaseTest.kt (hunk 8 above)
Comment (Mohammed-Alhams) [reply service_identity/src/test/kotlin/org/thechance/service_identity/domain/usecases/validation/UserValidationUseCaseTest.kt:13]: @Youssef-Ramadan-Y Have you completed the changes?

File: service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_management/IUserManagementUseCase.kt (hunk 2 above)
Comment (Mohammed-Alhams) [reply service_identity/src/main/kotlin/org/thechance/service_identity/domain/usecases/user_management/IUserManagementUseCase.kt]: Done

**Review by aya-seif — APPROVED** [review]

**Pull request merged**